zerver/lib/actions.py

```diff
@@ -2761,6 +2789,28 @@ def our_person(p):
                         bot.update(event['bot'])
 
         elif event['type'] == 'stream':
+            if event['op'] == 'create':
+                for stream in event['streams']:
+                    if not stream['invite_only']:
+                        stream_name = stream['name']
+                        realm = get_realm(user_profile.realm.domain)
+                        new_stream = get_stream(stream_name, realm)
+                        subscribers = get_subscriber_ids(new_stream)
+                        stream_data = {
+                            'description': new_stream.description,
+                            'stream_id': new_stream.id,
+                            'subscribers':  subscribers,
+                            'invite_only': new_stream.invite_only,
+                            'name': new_stream.name,
+                            'email_address': encode_email_address_helper(new_stream.name, new_stream.email_token)
+                        }
```

Comment:
`apply_events` should in theory not do any database queries, because the data it needs should be available in the events themselves.  In the case of stream create events, most of the data in your `stream_data` dict is inside the event (we have a stream.to_dict()), we probably don't need the stream's email address in the never_subscribed list...